[R-COMPAT] [q-mr1] Lose COPY_HEADERS, cleanup

ix5 opened the pull request:
This should not break your proprietaries, @jerpelea .

ix5 committed:
rmnetctl: Lose COPY_HEADERS, export instead

LOCAL_COPY_HEADERS is deprecated in R. Use
LOCAL_EXPORT_C_INCLUDE_DIRS and its equivalent in soong,
"export_include_dirs", instead.

Any proprietary module that links to rmnetctl will now
receive the headers in rmnetctl/inc/ automatically.

MarijnS95 commented on rmnetctl/src/Android.mk:
```diff
@@ -30,8 +31,7 @@ LOCAL_MODULE := librmnetctl
 
-LOCAL_MODULE_PATH_32 := $(TARGET_OUT_VENDOR)/lib
-LOCAL_MODULE_PATH_64 := $(TARGET_OUT_VENDOR)/lib64
+LOCAL_VENDOR_MODULE: true
```
```
vendor/qcom/opensource/dataservices/rmnetctl/src/Android.mk:34: error: writing to readonly directory: "LOCAL_VENDOR_MODULE"
```
You accidentally turned this one into a build rule :wink:

ix5 replied:
^^

ix5 replied:
Also, whaddaya think? https://github.com/ix5/vendor-qcom-opensource-dataservices/tree/q-mr1-nocopyheaders-androidbp

MarijnS95 replied:
You know I like `.bp` :smile: 

Let me apply some review inline as to not diverge it here.

ix5 committed:
treewide: Replace LOCAL_MODULE_PATH

Eh, it's cleaner to simply declare a vendor module. Also
helps with soong conversion.

MarijnS95 reviewed (approved):
Looks sane and compiles fine, can be merged IMO!

ix5 commented:
@jerpelea `:submarine-sonar-ping:`
